ADAL/src/ADAuthenticationContext+Internal.m

```diff
@@ -101,27 +101,35 @@ + (ADAuthenticationError*)errorFromDictionary:(NSDictionary*)dictionary
                                     errorCode:(ADErrorCode)errorCode
 {
     //First check for explicit OAuth2 protocol error:
-    NSString* serverOAuth2Error = [dictionary objectForKey:OAUTH2_ERROR];
+    NSString *serverOAuth2Error = [dictionary objectForKey:OAUTH2_ERROR];
     if (![NSString adIsStringNilOrBlank:serverOAuth2Error])
     {
-        NSString* errorDetails = [dictionary objectForKey:OAUTH2_ERROR_DESCRIPTION];
+        NSString *errorDetails = [dictionary objectForKey:OAUTH2_ERROR_DESCRIPTION];
         // Error response from the server
-        NSUUID* correlationId = [dictionary objectForKey:OAUTH2_CORRELATION_ID_RESPONSE] ?
+        NSUUID *correlationId = [dictionary objectForKey:OAUTH2_CORRELATION_ID_RESPONSE] ?
                                 [[NSUUID alloc] initWithUUIDString:[dictionary objectForKey:OAUTH2_CORRELATION_ID_RESPONSE]]:
                                 nil;
 
         ADErrorCode code = errorCode;
-        NSString* suberror = [dictionary objectForKey:AUTH_SUBERROR];
+        NSString *suberror = [dictionary objectForKey:AUTH_SUBERROR];
+        NSMutableDictionary *userInfo = nil;
         if (suberror && [suberror isEqualToString:AUTH_PROTECTION_POLICY_REQUIRED])
         {
             code = AD_ERROR_SERVER_PROTECTION_POLICY_REQUIRED;
+            userInfo = [[NSMutableDictionary alloc] initWithCapacity:2];
+            [userInfo setObject:suberror forKey:ADSuberrorKey];
+
+            // check for additional user identifier
+            NSString* userId = [dictionary objectForKey:AUTH_ADDITIONAL_USER_IDENTIFIER];
+            if (userId)
+                [userInfo setObject:userId forKey:ADUserIdKey];
```

Comment:
nit: just use subscripts, so there's no need to check if it's non nil
e.g. userInfo[ADUserIdKey] = userId